### PR TITLE
Remove API that was deprecated during the 5.10 release

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: 

## Summary

Now that version 6.0 is released, API that's been deprecated during the 5.10 release can be removed according to the process described in the "Introducing source breaking changes" section of CONTRIBUTING.md.

## Dependencies

None.

## Testing

Nothing in particular. DocC itself wasn't using this API anymore.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- ~[ ] Added tests~
- [x] Ran the `./bin/test` script and it succeeded
- ~[ ] Updated documentation if necessary~

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
This repo doesn't have a 6.0 release yet [but Swift has](https://github.com/swiftlang/swift/releases/tag/swift-6.0-RELEASE).

File: Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsumer.swift [hunk 1]
```diff
@@ -16,10 +16,6 @@ public protocol DiagnosticConsumer: AnyObject {
     /// - Parameter problems: The encountered diagnostics.
     func receive(_ problems: [Problem])
     
-    /// Inform the consumer that the engine has sent all diagnostics for this build.
```
Comment (patshaughnessy) [inline Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsumer.swift:19]: Let's also update the copyright message for this file to 2024.

File: Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticEngine.swift [hunk 2]
```diff
@@ -105,11 +105,6 @@ public final class DiagnosticEngine {
         }
     }
     
-    @available(*, deprecated, renamed: "flush()", message: "Use 'flush()' instead. This deprecated API will be removed after 6.0 is released")
```
Comment (patshaughnessy) [inline Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticEngine.swift:108]: Same as above: update the copyright message to 2024.

File: Sources/SwiftDocCUtilities/ArgumentParsing/Options/DocumentationBundleOption.swift [hunk 3]
```diff
@@ -25,6 +25,3 @@ public struct DocumentationCatalogOption: DirectoryPathOption {
         transform: URL.init(fileURLWithPath:))
     public var url: URL?
 }
-
```
Comment (patshaughnessy) [inline Sources/SwiftDocCUtilities/ArgumentParsing/Options/DocumentationBundleOption.swift:28]: Same as above: update the copyright message to 2024.

**Review by patshaughnessy — APPROVED** [review]
Looks good - maybe just update the copyright message on some modified source files.

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**